applications/MeshingApplication/tests/cpp_tests/test_metric_process.cpp

```diff
@@ -400,159 +390,178 @@ namespace Kratos
+        }
+        
+        /** 
+        * Checks the correct work of the nodal SPR compute
+        * Test tetrahedra
+        */
+
+        KRATOS_TEST_CASE_IN_SUITE(TestSPRMetricProcess2, KratosMeshingApplicationFastSuite)
+        {
+            ModelPart this_model_part("Main");
+            this_model_part.SetBufferSize(2);
             
-            std::vector<NodeType::Pointer> element_nodes_10 (4);
-            element_nodes_10[0] = p_node_9;
-            element_nodes_10[1] = p_node_12;
-            element_nodes_10[2] = p_node_11;
-            element_nodes_10[3] = p_node_8;
-            Tetrahedra3D4 <NodeType> tetrahedra_10( PointerVector<NodeType>{element_nodes_10} );
+            this_model_part.AddNodalSolutionStepVariable(NODAL_H);
+            this_model_part.AddNodalSolutionStepVariable(DISPLACEMENT);
             
-            std::vector<NodeType::Pointer> element_nodes_11 (4);
-            element_nodes_11[0] = p_node_3;
-            element_nodes_11[1] = p_node_2;
-            element_nodes_11[2] = p_node_1;
-            element_nodes_11[3] = p_node_6;
-            Tetrahedra3D4 <NodeType> tetrahedra_11( PointerVector<NodeType>{element_nodes_11} );
+            auto& process_info = this_model_part.GetProcessInfo();
+            process_info[STEP] = 1;
+            process_info[NL_ITERATION_NUMBER] = 1;
             
-            Element::Pointer p_elem_0 = this_model_part.CreateNewElement("Element3D4N", 1, tetrahedra_0, p_elem_prop);
-            Element::Pointer p_elem_1 = this_model_part.CreateNewElement("Element3D4N", 2, tetrahedra_1, p_elem_prop);
-            Element::Pointer p_elem_2 = this_model_part.CreateNewElement("Element3D4N", 3, tetrahedra_2, p_elem_prop);
-            Element::Pointer p_elem_3 = this_model_part.CreateNewElement("Element3D4N", 4, tetrahedra_3, p_elem_prop);
-            Element::Pointer p_elem_4 = this_model_part.CreateNewElement("Element3D4N", 5, tetrahedra_4, p_elem_prop);
-            Element::Pointer p_elem_5 = this_model_part.CreateNewElement("Element3D4N", 6, tetrahedra_5, p_elem_prop);
-            Element::Pointer p_elem_6 = this_model_part.CreateNewElement("Element3D4N", 7, tetrahedra_6, p_elem_prop);
-            Element::Pointer p_elem_7 = this_model_part.CreateNewElement("Element3D4N", 8, tetrahedra_7, p_elem_prop);
-            Element::Pointer p_elem_8 = this_model_part.CreateNewElement("Element3D4N", 9, tetrahedra_8, p_elem_prop);
-            Element::Pointer p_elem_9 = this_model_part.CreateNewElement("Element3D4N", 10, tetrahedra_9, p_elem_prop);
-            Element::Pointer p_elem_10 = this_model_part.CreateNewElement("Element3D4N", 11, tetrahedra_10, p_elem_prop);
-            Element::Pointer p_elem_11 = this_model_part.CreateNewElement("Element3D4N", 12, tetrahedra_11, p_elem_prop);
+            // In case the StructuralMechanicsApplciation is not compiled we skip the test
```

Comment:
how are u achieving to skip the test. Wouldn't it be better to have this test somewhere else? it checks a different feature...